services/web/server/src/simcore_service_webserver/exporter/request_handlers.py

```diff
@@ -38,24 +42,38 @@ async def export_project(request: web.Request):
     temp_dir: str = await get_empty_tmp_dir()
 
     try:
-        file_to_download = await study_export(
-            app=request.app,
-            tmp_dir=temp_dir,
-            project_id=project_uuid,
-            user_id=user_id,
-            product_name=request[RQ_PRODUCT_KEY],
-            archive=True,
-        )
-        log.info("File to download '%s'", file_to_download)
-
-        if not file_to_download.is_file():
-            raise ExporterException(
-                f"Must provide a file to download, not {str(file_to_download)}"
+        async with await lock_project(
+            request.app,
+            project_uuid,
+            ProjectStatus.EXPORTING,
+            user_id,
+            await get_user_name(request.app, user_id),
+        ):
+            await retrieve_and_notify_project_locked_state(
+                user_id, project_uuid, request.app
+            )
```

Comment:
Why is this not done inside the context manager, maybe by providing an option to it?

Reply:
Yes, I'd like that too. I'll see if python import mechanism allows that cause this will create a circular import. I'll check it out.

Reply:
will add it once py3.8 is in